util/random/rand.go

```diff
@@ -71,8 +76,60 @@ func (r *randstream) XORKeyStream(dst, src []byte) {
 	}
 }
 
-// New returns a new cipher.Stream that gets random data from Go's crypto/rand package.
+// READER_BYTES is how many bytes we expect from each source
+const READER_BYTES = 32
+
+// MixedrandHash is the hash used in XORKeyStream to mix all entropy sources together.
+// sha512 is chosen as its digest is 64 bytes long, matching the length of blake2xb's seed.
+var MixedrandHash = sha512.New
```

Comment:
It is unclear why this is here; I can see that a caller could choose to modify it, but the docs don't explain that. And don't explain the thread-safety implications of modifying it. I don't see the interest in being able to modify this, let's just hard code it for now.

Reply:
* Good point about the thread safety. 
* About the doc mentionning the change, well, the doc can be changed ;) 
* It's not so uncommon to have variables that can be changed globally in Go code but have a good default purpose. See for example the `crypto/rand.Reader` variable: https://golang.org/pkg/crypto/rand/#pkg-variables . It's even worse in that case because the global variable is mutable. 
* In any case, after some thinking, I would much prefer to stick with `SHA256` (or even better  Blake2s). No need to inccur additional performance hit for no reason (SHA3 is much slower than the two other).

Reply:
So you would prefer this line https://github.com/dedis/kyber/blob/93efc4f605eccd9d74cb6de04a9f67b0bb5e0ab3/util/random/rand.go#L116 to simply be `h := sha256.New()` ?